src/Trigger/BasicDeliverEventArgsValueProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Newtonsoft.Json;
+using RabbitMQ.Client.Events;
+
+namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ
+{
+    public class BasicDeliverEventArgsValueProvider : IValueProvider
+    {
+        private readonly BasicDeliverEventArgs _input;
+
+        public BasicDeliverEventArgsValueProvider(BasicDeliverEventArgs input, Type destinationType)
+        {
+            _input = input;
+            Type = destinationType;
+        }
+
+        public Type Type { get; }
+
+        public Task<object> GetValueAsync()
+        {
+            if (Type.Equals(typeof(BasicDeliverEventArgs)))
+            {
+                return Task.FromResult<object>(_input);
+            }
+
+            string inputValue = ToInvokeString();
+            if (Type.Equals(typeof(string)))
+            {
+                return Task.FromResult<object>(inputValue);
+            }
+            else
```

Comment:
This else block is just for Poco handling? How/where is JObject handled?

Reply:
Yes. This is handled in dotnet scenario where data is being bound to a Poco like in a below scenario

````
public class TestClass
    {
        public string X { get; set; }
    }

    public static class Function1
    {
        [FunctionName("Function1")]
        public static void Run([RabbitMQTrigger("TestQueue", ConnectionStringSetting = "ConnectionString")] TestClass myQueueItem, ILogger log)
        {
            log.LogInformation($"C# Queue trigger function processed: {myQueueItem}");
        }
    }
````